lib/fetch/body.js

```diff
@@ -397,7 +398,40 @@ function bodyMixinMethods (instance) {
 
       // If mimeType’s essence is "multipart/form-data", then:
       if (/multipart\/form-data/.test(contentType)) {
-        throw new NotSupportedError('multipart/form-data not supported')
+        const headers = {}
+        for (const [key, value] of this.headers) headers[key.toLowerCase()] = value
+
+        const responseFormData = new FormData()
+
+        try {
+          const busboy = Busboy({ headers })
+          busboy.on('field', (name, value) => {
+            responseFormData.append(name, value)
+          })
+          busboy.on('file', (name, value, info) => {
+            const { filename, encoding, mimeType } = info
+            const base64 = encoding.toLowerCase() === 'base64'
+            const chunks = []
+            value.on('data', (chunk) => {
+              if (base64) chunk = Buffer.from(chunk.toString(), 'base64')
+              chunks.push(chunk)
```

Comment:
This will buffer a lot of data in memory. It needs some form of limit or backpressure.

Reply:
I don't think we can easily implement or put a limit. That's the fundamental weakness of parsing multipart with `fetch()`. We should document this.

The actual solution is to have disk-backed `Blob` objects, but that would need to be implemented in Node.js core. There is only so much we can do ourselves here.

Reply:
Think this is good as a MVP for now. At least it will be spec compatible and also making it at least possible to use `.fromData()` for those who might need it. It will at least solve the problem for ppl who might just handle some small tiny files (like profile picture uploads). or just only using simple `<form>`s without any files and still uses `new FormData()` to upload just text inputs as multipart payloads.

I agree with @mcollina that disc-backed `Blob`s are necessary. 
Think we can try to improve this later once disc-backed `Blob`s are in place. 
But right now I just wish `res.formData()` to at least work, even if it meant that it was handled badly. 

---

I think and i believe whenever you fetch something with either `xhr.responseType = 'blob'` or do `response.blob()` in browser that they just do pipe the data to the disc rather than reading it all to the memory, but i'm not sure... Browser can do all kind of magic to blobs. And they might all works differently in all browsers. They can try to offload some of "in memory blobs" to the disc if something is taking up too much memory. 

i have tried to make since of this old [presentation](https://docs.google.com/presentation/d/1MOm-8kacXAon1L2tF6VthesNjXgx0fp5AP17L7XDPSM/edit#slide=id.g75d5729ce_0_145) when building `fetch-blob`

---

Is there any reason why we can't just bring in the `Blob` and `File` class from v8 itself instead of building our own kind of polyfilled File/Blob class written in javascript?

think Bun.js was able to do this with JavaScriptCore (webkit). 
So when you pass a file/blob to `Bun.write(blob)`, Bun automatically uses a faster system call:
```js
const blob = Bun.file("input.txt");
await Bun.write("output.txt", blob);
```
On Linux, this uses the [copy_file_range](https://man7.org/linux/man-pages/man2/copy_file_range.2.html) syscall and on macOS, this becomes clonefile (or [fcopyfile](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/copyfile.3.html)).

Reply:
> Is there any reason why we can't just bring in the Blob and File class from v8 itself instead of building our own kind of polyfilled File/Blob class written in javascript?

V8 does not have them, they are from Chromium. They should be implemented in Node.js.

Reply:
Should this thread be a discussion?